pkg/validator/config.go

```diff
@@ -1,15 +1,16 @@
 package validator
 
 import (
+	"errors"
 	"fmt"
 	"maps"
 
 	"github.com/fusakla/promruval/v3/pkg/config"
+	"github.com/fusakla/promruval/v3/pkg/unmarshaler"
 	"github.com/prometheus/prometheus/model/rulefmt"
-	"gopkg.in/yaml.v3"
 )
 
-type validatorCreator func(params yaml.Node) (Validator, error)
+type validatorCreator func(unmarshal func(interface{}) error) (Validator, error)
```

Comment:
Might be worth making it a type, looking at the number of occurrences?
```suggestion
type unmarshalParamsFunc func(interface{}) error
type validatorCreator func(unmarshal unmarshallParamsFunc) (Validator, error)
```